app/updatemanager.go

```diff
@@ -149,9 +150,22 @@ type UpdateControlMapState struct {
 }
 
 type UpdateControlMap struct {
-	ID       string                           `json:"id"`
-	Priority int                              `json:"priority"`
-	States   map[string]UpdateControlMapState `json:"states"`
+	ID         string                           `json:"id"`
+	Priority   int                              `json:"priority"`
+	States     map[string]UpdateControlMapState `json:"states"`
+	expiryTime time.Time
+	expired    bool
+}
+
+func (u *UpdateControlMap) Stamp(updateControlTimeoutSeconds int) *UpdateControlMap {
+	u.expiryTime = time.Now().Add(
+		time.Duration(updateControlTimeoutSeconds) * time.Second)
+	// expiration monitor
+	go func() {
+		<-time.After(time.Duration(updateControlTimeoutSeconds) * time.Second)
+		u.expired = true
```

Comment:
This doesn't really solve the problem, unfortunately. The issue is that the event loop needs to know when an expiration has happened, and then reevaluate whether it should stay in the current state or not.

But I think I would recommend that you just leave this alone for now. What you have works for the current acceptance criteria, and it should be clearer exactly what you need once you start working on MEN-4549.

Reply:
Postponing it till `MEN-4549` :)

Then it'll be either mine, or urs problem :p